### PR TITLE
fix(textbox, select): updated colors to match design system

### LINKED ISSUE #1809: Textbox component colors are inconsistent with the Figma design system
- Readonly Textboxes have no difference in colors from normal Textboxes
    image
- Error-State Textboxes have a red background instead of red text
    image
- The background color of Readonly Textboxes is updated when selected, which gives the impression that users can type

### PR DESCRIPTION
**Pull request by LuLaValva**
- Fixes #1809 

- [x] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description

- Match colors to design system for:
  - readonly textbox & select
  - disabled textbox & select
  - error-state disabled & select
- Prevented readonly textbox & select from changing colors on focus

## Notes

- In `textbox.less`, I was not able to make a commit until I moved the `textarea` and `input` focus outside of the nested component. It may be harder to find, but the linter likes it more.
- The scope of this issue increased as I was working on it, when I recognized more portions of the component which were not consistent with the design system.

### Before:
image
image
image
image

### After:
image
image
image
image

## Checklist

- [x] I verify the build is in a non-broken state
- [x] I verify all changes are within scope of the linked issue

- [x] I regenerated all CSS files under dist folder
- [x] I tested the UI in all supported browsers
- [x] I tested the UI in dark mode and RTL mode
- [ ] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

**Comment by LuLaValva** [comment]
> Just make sure the dark mode colors in Figma match the dark mode vars we have.

Thanks @ArtBlue, I've updated the dark mode colors accordingly.

**Review by agliga — COMMENTED** [review]
Couple of issues:

1. Figma has readonly without any boarders and only shows the text floating.
2. Outline is being shown on readonly when you tab through. 
Again, I think the only thing we probably should change with this is the input color. If we want to update to match design, we should update it all.
Screen Shot 2022-07-20 at 4 20 21 PM
Screen Shot 2022-07-20 at 4 21 24 PM

**Comment by LuLaValva** [comment]
> Figma has readonly without any boarders and only shows the text floating.

I see now! I was looking at the wrong row in Figma, for some reason I didn't think to look on the left side. It has been fixed.

> I think the only thing we probably should change with this is the input color.

Ok, do you think I should take out the changes that remove selection on readonly?

It is worth noting that the HTML specs do not actually support `readonly` on `select` components. These are still fully editable by the user, tested in Chrome. A suggested way to fix this is to replace readonly `select` components with disabled ones, and add actual components to act on the form that are hidden.

File: src/less/textbox/textbox.less [hunk 1]
```diff
@@ -133,6 +125,14 @@ input.textbox__control--large {
     }
 }
 
+input.textbox__control:not(:read-only):focus,
+textarea.textbox__control:not(:read-only):focus {
+    .border-color-token(textbox-focus-border-color, color-stroke-default);
+    .background-color-token(textbox-focus-background-color, color-background-primary);
+    .color-token(textbox-focus-foreground-color, color-foreground-secondary);
```
Comment (agliga) [inline src/less/textbox/textbox.less:132]: I believe this color is wrong.
In fact it should just not even be set, the color of the focused text should be the same as the unfocused.

**Comment by agliga** [comment]
> > Figma has readonly without any boarders and only shows the text floating.
> 
> I see now! I was looking at the wrong row in Figma, for some reason I didn't think to look on the left side. It has been fixed.
> 
> > I think the only thing we probably should change with this is the input color.
> 
> Ok, do you think I should take out the changes that remove selection on readonly?
> 
> It is worth noting that the HTML specs do not actually support `readonly` on `select` components. These are still fully editable by the user, tested in Chrome. A suggested way to fix this is to replace readonly `select` components with disabled ones, and add actual components to act on the form that are hidden from the user.

I think the PR is overall fine. 
I think you should pull from `14.1.0` and resolve the conflicts. 
The color you have is off on focus, but if you remove that (which is what the merge conflict is), then it should be fine. 
Everything else looked great to me.

**Review by agliga — APPROVED** [review]

**Pull request merged**